### DISCUSSION TITLE
Are there plans to integrate with the new testing API?

### DISCUSSION THREAD

**Discussion by pmcelhaney** [orta/vscode-jest#751] [Q&A] [answered]
Edit: yes. 

https://github.com/jest-community/vscode-jest/pull/742

The July 2021 release of VS Code has new language-agnostic [testing APIs](https://code.visualstudio.com/updates/v1_59#_testing-apis) and UIs that are clearly inspired by this extension. 

> Last fall we began work on adding native support for running tests in VS Code, and this month the first set of testing-related APIs has been finalized. These APIs provide greater flexibility, better performance, and richer UX over what has been found previously in extensions. Check out the guide on writing testing extensions to dive in.

Are there plans for vscode-jest to take advantage of the new API? There's [another extension](https://marketplace.visualstudio.com/items?itemName=kavod-io.vscode-jest-test-adapter) out there that already does that but this one seems more mature.

**Comment by connectdotz** [comment, marked as answer]
indeed, feel free to give the rc version a try and let us know how it works for you

**Reply by JustinGrote** [reply]
@connectdotz works quite well for me! Would be nice if the errors were surfaced for the expected/actual. Here's an example from my Pester Test adapter for what it could look like, feel free to steal my code examples, the vscode markdown example also shows it.
https://github.com/pester/vscode-adapter/blob/main/images/2021-08-07-08-06-26.png

https://github.com/pester/vscode-adapter

**Comment by connectdotz** [comment]
inlineError is indeed supported, you can turn it on via [jest.testExplorer](https://github.com/jest-community/vscode-jest#testexplorer), by `setting "showInlineError": true`

**Reply by JustinGrote** [reply]
Ah thanks curious why do the options this way instead of jest.testExplorer.showinelineError? I couldn't discover it via settings intellisense because it's just one object.

**Comment by connectdotz** [comment]
@JustinGrote by making them separate settings, there is no way to prevent invalid combinations, such as: 
```
"jest.testExplorer.enabled": false
"jest.testExplorer.showinelineError": true
```
Given Typescript language service already supports the discovery for this kind of type, I hope the setting editor can be improved accordingly.
